DogScepterLib/Core/GMData.cs

```diff
@@ -4,109 +4,195 @@
-                        }
-                    }
-                }
+                Minor = minor;
+                Release = release;
+                Build = build;
+                return;
             }
+            if (Minor > minor) return;
 
-            /// <summary>
-            /// Returns whether the recorded version number is greater than or equal to the parameters
-            /// </summary>
-            public bool IsNumberAtLeast(int major = 0, int minor = 0, int release = 0, int build = 0)
+            // Our Minor is now equal to proposed minor.
+            if (Release < release)
             {
-                if (Major != major)
-                    return (Major > major);
-                if (Minor != minor)
-                    return (Minor > minor);
-                if (Release != release)
-                    return (Release > release);
-                if (Build != build)
-                    return (Build > build);
-                return true;
+                Release = release;
+                Build = build;
+                return;
             }
+            if (Release > release) return;
+
+            // Our Release is now equal to proposed release.
+            if (Build < build)
+                Build = build;
         }
 
-        public class GMStats
+        /// <summary>
+        /// Returns whether the recorded version number is greater than or equal to the parameters.
+        /// </summary>
+        public bool IsNumberAtLeast(int major = 0, int minor = 0, int release = 0, int build = 0)
         {
-            public int LastStructID = -1;
+            if (Major != major)
+                return (Major > major);
+            if (Minor != minor)
+                return (Minor > minor);
+            if (Release != release)
+                return (Release > release);
+            if (Build != build)
+                return (Build > build);
+            return true;
         }
+    }
 
-        public GMVersionInfo VersionInfo;
-        public GMStats Stats;
-
-        public GMChunkFORM FORM;
-        public Dictionary<string, GMChunk> Chunks => FORM.Chunks;
-#if DEBUG
-        public Log Logger = Console.WriteLine;
-#else
-        public Log Logger = null;
-#endif
+    //TODO:?
+    public class GMStats
+    {
+        public int LastStructID = -1;
+    }
 
-        // Handles writing miscellaneous files asynchronously
-        public ActionBlock<KeyValuePair<string, BufferRegion>> FileWrites = new ActionBlock<KeyValuePair<string, BufferRegion>>(f =>
+    /// <summary>
+    /// Contains various version information of this data file.
+    /// </summary>
+    public GMVersionInfo VersionInfo;
+
+    public GMStats Stats;
+
+    /// <summary>
+    /// The FORM chunk of the data file.
+    /// </summary>
+    public GMChunkFORM FORM;
+
+    /// <summary>
+    /// All chunks of the data file.
+    /// </summary>
+    //TODO: make this <GMChunkNames, GMChunk> in order to mitigate typos?
```

Comment:
Unsure what the point of abstracting the name out would be. It's not like we're adding new chunks all the time, and if there's a typo, we'll know immediately, because the code just won't work.

Reply:
It'd move the time window from "the code just won't work" from runtime to compile time. 
It might happen, that when adding new code, that it's forgotten to check if the added code works, or that there are no tests to cover that case. 
Having these be abstracted, would ensure that the compiler / IDE screams when trying to compile, which would make it easier to catch errors.

Reply:
Yeah, I get that it would make it a compile time error... but these are the _main_ chunks here. If one doesn't load properly, you literally can't test anything using that chunk, which would be an obvious error at runtime anyway, because you'd get an unrecognized chunk warning/error. If it makes the code cleaner I'm fine with it, though, but I'm not sure it will.

Reply:
Well, I meant this more for cases where something along the lines of `var locals = Data.Chunks["FUNC"]).Locals` is used. 
I'll remove the todo for now, and do a seperate PR for this that tries to implement this.

Reply:
Ah, you know, that case is actually fair.